tools/xtb/xtb_molecular_optimization.xml

```diff
@@ -24,29 +35,48 @@
             <option value="vtight">vtight</option>
             <option value="extreme">extreme</option>
         </param>
+        <param type="boolean" name="replace_second_line" label = "Keep molecule name" truevalue="TRUE" falsevalue="FALSE" checked="false" help = "If YES, the molecule name from the coordinates file is preserved, if it exists. Otherwise, the energy, gnorm, and xtb version are printed." />
     </inputs>
     <outputs>
         <data name="output" format_source="coordinates"  />
     </outputs>
     <tests>
         <test>
-            <param name="coordinates" value="initial_mol.xyz" ftype="xyz"/>
+            <param name="coordinates" value="initial_mol_molname.xyz" ftype="xyz"/>
+            <param name="replace_second_line" value="TRUE" />
             <output name="output">
                 <assert_contents>
-                    <has_text text="xtb"/>
-                    <has_n_lines n="26"/> 
+                    <has_text text="Benzophenone"/>
                 </assert_contents>
             </output>
         </test>
-        <!-- <test>
-            <param name="coordinates" value="initial_mol.coord" ftype="coord"/>
-            <output name="output" file="optimized_mol.coord" ftype="coord">
+        <test>
+            <param name="coordinates" value="initial_mol_molname.xyz" ftype="xyz"/>
+            <param name="replace_second_line" value="FALSE" />
+            <output name="output">
+                <assert_contents>
+                    <has_text text="energy"/>
+                </assert_contents>
+            </output>
+        </test>
+        <test>
+            <param name="coordinates" value="initial_mol_no_molname.xyz" ftype="xyz"/>
+            <param name="replace_second_line" value="TRUE" />
+            <output name="output">
                 <assert_contents>
-                    <has_text text="$coord"/>
-                    <has_n_lines min="26"/> 
+                    <has_text text=" "/>
```

Comment:
This merely tests if it has a space. There is I think an assert to check a the content of a specific line - this could be useful here.

Reply:
I think the previous check was maybe even more concise in this place.

Reply:
I think <not_has_text text="energy"/> fits to this case as well [b4eb0e6](https://github.com/RECETOX/galaxytools/pull/557/commits/b4eb0e6309b132c5da5bc80e11322cc01b12b454).